napari/layers/labels/_tests/test_labels.py

```diff
@@ -263,26 +285,28 @@ def test_seed():
 
 def test_num_colors():
     """Test setting number of colors in colormap."""
-    np.random.seed(0)
-    data = np.random.randint(20, size=(10, 15))
-    layer = Labels(data)
-    assert layer.num_colors == 49
+    with pytest.warns(FutureWarning, match='num_colors is deprecated'):
+        np.random.seed(0)
+        data = np.random.randint(20, size=(10, 15))
+        layer = Labels(data)
+        assert layer.num_colors == 50
 
-    layer.num_colors = 80
-    assert layer.num_colors == 80
+        layer.num_colors = 80
+        assert layer.num_colors == 80
 
-    layer = Labels(data, num_colors=60)
-    assert layer.num_colors == 60
+        layer = Labels(data, num_colors=60)
+        assert layer.num_colors == 60
 
-    with pytest.raises(
-        ValueError, match=r".*Only up to 2\*\*16=65535 colors are supported"
-    ):
-        layer.num_colors = 2**17
+        with pytest.raises(
+            ValueError, match=r".*Only up to 2\*\*16=65535 colors"
+        ):
+            layer.num_colors = 2**17
```

Comment:
only line triggering warning should be under `pytest.warns`. So it means that there should be multiple `with pytest.warns`